components/server/src/auth/generic-auth-provider.ts

```diff
@@ -258,7 +267,17 @@ export abstract class GenericAuthProvider implements AuthProvider {
         });
 
         const isAlreadyLoggedIn = request.isAuthenticated() && User.is(request.user);
-        const authFlow = AuthFlow.get(request.session);
+
+        const state = request.query.state;
+        if (!state) {
+            log.error(cxt, `(${strategyName}) No state present on callback request.`, { clientInfo });
+            response.redirect(
+                this.getSorryUrl(`No state was present on the authentication callback. Please try again.`),
+            );
+            return;
+        }
+
+        const authFlow = await this.parseState(state as string);
```

Comment:
Potential parse/verification errors should be caught here, redirecting to the default error page would be ok. 

To reproduce a dangling request, first log out, then navigate to https://github.com/login/oauth/authorize?response_type=code&redirect_uri=http%3A%2F%2Fpreview.gitpod-dev.com%2Fauth%2Fgithub%2Fcallback&scope=user%3Aemail&state=mp-server-70f37720f6%2CeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9-bar&client_id=6025fa48a14288c318ad